react-app/src/navigation/AppRouter.tsx

```diff
@@ -1,14 +1,28 @@
 import React from "react";
+import { useQuery } from "@apollo/client";
 import { BrowserRouter, Route, Routes } from "react-router-dom";
 import AppSideBar from "../components/AppSideBar/AppSideBar";
 import DummyScreen from "../components/DummyScreen/DummyScreen";
 import OverviewScreen from "../components/OverviewScreen/OverviewScreen";
+import { ChainHealth, ChainHealthQuery } from "../generated/graphql";
 import AppRoutes from "./AppRoutes";
 
 const AppRouter: React.FC = () => {
+  const { loading, data, error } = useQuery<ChainHealthQuery>(ChainHealth);
```

Comment:
Why query in router level?
If these will be reused in deeper components, make a context provider;
Otherwise if only side bar uses this then move to AppSIdeBar?

Reply:
For screen queries let's name them AbcScreenQuery; and make a AbcScreenAPI.ts;

Let's also introduce RequestState? so no repeated checks for loading / error